### ISSUE TITLE
Creating data values on object after deletion of another object

### ISSUE THREAD

**Issue by maxiundtesa** [realm/realm-cocoa#3601]
Hello,
I am having the problem, that when I am deleting a realm object, it is deleting the object but at the same time it is modifying another object.
This image is from the Realm Browser before i delete an object.

![realm-vorher](https://cloud.githubusercontent.com/assets/17526240/15327654/84e6b966-1c52-11e6-98c8-f8ea91e134f6.png)

This image is after i deleted it.
![realm-nachher](https://cloud.githubusercontent.com/assets/17526240/15327677/96df4c5a-1c52-11e6-81f2-22e5787b7cda.png)

You can see that the data properties, which were nil before, are now having a data value.
I am using Swift, realm 102.1,  Xcode 7.3.1 and the simulator is running iOS 9.3

This is my realm object class and the delete Method
```
import Foundation
import RealmSwift

class MessageRealm: Object {
    dynamic var absenderId: String? = nil
    dynamic var absenderName: String? = nil
    dynamic var datum: NSDate? = nil
    dynamic var gruppenId: String? = nil
    dynamic var image: NSData? = nil
    dynamic var latitude = 0.0
    dynamic var longitude = 0.0
    dynamic var messageId: String? = nil
    dynamic var uebergreifendId : String? = nil
    dynamic var question: String? = nil
    dynamic var sound: NSData? = nil
    dynamic var video: NSData? = nil
    dynamic var livePhoto: NSData? = nil
    dynamic var text: String? = nil
    
    dynamic var wurdeVerschickt: Bool = false
    let runtergeladenAngeschaut = List<RuntergeladenRealm>()
    let answers = List<WertRealm>()
    let results = List<WertRealm>()
    let userIDs = List<WertRealm>()

    override class func primaryKey() -> String {
        return "messageId"
    }

}
```

```

override func collectionView(collectionView: JSQMessagesCollectionView!, didDeleteMessageAtIndexPath indexPath: NSIndexPath!) {
        let message = messages[indexPath.row]

        let nachricht = realm.objectForPrimaryKey(MessageRealm.self, key: message.messageId)
        if nachricht != nil {
            try! self.realm.write({ () -> Void in
                realm.delete(nachricht!)
            })
            
            messages.removeAtIndex(indexPath.row)
            
            
        }
        else {
            messages.removeAtIndex(indexPath.row)
            
        }
        
    }
```

**Comment by mrackwitz** [comment]
I can't load the attached screenshot what makes it hard to follow the error report. Can you reproduce this error? Do you think you could try to isolate it and share a sample project with us? That would help a lot.

**Comment by maxiundtesa** [comment]
This is the image before the deletion: https://onedrive.live.com/redir?resid=C126B3280851C3BC!1141&authkey=!AA8J6Pyd52jmm4Y&v=3&ithint=photo%2cpng

This is the image after the deletion : https://onedrive.live.com/redir?resid=C126B3280851C3BC!1142&authkey=!AEt-1e8BSf152FE&v=3&ithint=photo%2cpng

And this is the sample project to reproduce the problem: 
https://onedrive.live.com/redir?resid=C126B3280851C3BC!1143&authkey=!AHD-kz16L0LTKc8&ithint=file%2czip

**Comment by tgoyne** [comment]
Thanks for the repro case. I've submitted a fix for this in the core library.
